modules/cssn/src/Plugin/Util/ProjectLookup.php

```diff
@@ -0,0 +1,154 @@
+    $query->fields('wsd', ['sid', 'name']);
+    $query->condition($orGroup);
+    $query->condition('wsd.webform_id', 'project');
+    $result = $query->execute()->fetchAll();
+
+    $query_flag = \Drupal::database()->select('flagging', 'f');
+    $query_flag->fields('f', ['entity_id', 'flag_id']);
+    $query_flag->condition('f.uid', $project_user_id);
+    $query_flag->condition('f.flag_id', 'interested_in_project');
+    $result_flag = $query_flag->execute()->fetchAll();
+    $flagged_results = array_map(function($result_flag) {
+      return (object) [
+        'sid' => $result_flag->entity_id,
+        'name' => 'interested_in_project',
+      ];
+    }, $result_flag);
+    $result = array_merge($result, $flagged_results);
+    if ($result != NULL) {
+      foreach ($result as $project_result) {
+        $wf = WebformSubmission::load($project_result->sid);
+        $wf_lookup = $wf->getData();
+        $this->projects[] = [
+          'title' => $wf_lookup['project_title'],
+          'name' => $project_fields[$project_result->name],
+          'status' => $wf_lookup['status'],
+          'sid' => $project_result->sid,
+        ];
+      }
+    }
+  }
+
+  /**
+   * Function to sort by status.
+   */
+  public function sortStatusProjects() {
+    $projects = $this->projects;
+    $recruiting = $this->arrayPickSort($projects, 'Recruiting');
+    $in_progress = $this->arrayPickSort($projects, 'In Progress');
+    $in_review = $this->arrayPickSort($projects, 'Reviewing Applicants');
+    $on_hold = $this->arrayPickSort($projects, 'On Hold');
+    $finishing_up = $this->arrayPickSort($projects, 'Finishing Up');
+    $complete = $this->arrayPickSort($projects, 'Complete');
+    $halted = $this->arrayPickSort($projects, 'Halted');
+    // Combine all of the arrays.
+    $projects_sorted = $recruiting + $in_progress + $in_review + $on_hold + $finishing_up + $complete + $halted;
+    $this->projects_sorted = $projects_sorted;
+  }
+
+  /**
+   * Function to pick out a status into an array and sort by title.
+   */
+  public function arrayPickSort($array, $sortby) {
+    $sorted = [];
+    if ($array == NULL) {
+      return;
+    }
+    foreach ($array as $key => $value) {
+      if ($value['status'] && $value['status'] == $sortby) {
+        $sorted[$key] = $value;
+      }
+    }
+    return $sorted;
+  }
+
+  /**
+   * Function to return styled list.
+   */
+  public function getProjectList() {
+    $n = 1;
+    $project_link = '';
+    if ($this->projects_sorted == NULL) {
+      return;
+    }
+    foreach ($this->projects_sorted as $project) {
+      $stripe_class = $n % 2 == 0 ? 'bg-light' : '';
+      $title = $project['title'];
+      $sid = $project['sid'];
+      $project_status = $project['status'];
+      $project_name = $project['name'];
+      if (($project_status == 'Recruiting' && $project_name == 'Interested') || $project_name != 'Interested') {
+        $lowercase = lcfirst($project_name);
```

Comment:
Seeing warnings like this:
```Deprecated function: lcfirst(): Passing null to parameter #1 ($string) of type string is deprecated in Drupal\cssn\Plugin\Util\ProjectLookup->getProjectList() (line 125 of modules/custom/access/modules/cssn/src/Plugin/Util/ProjectLookup.php).```
The null $project_name is also breaking the Font Awesome icon class.
The conditions on line 124 seem strange also. Seems like interested should be a status?

Reply:
I can't see any of these issues you're listing here. Are you testing as a specific user?